pkg/apiserver/rest/dualwriter.go

```diff
@@ -95,20 +98,31 @@ const (
 	Mode4
 )
 
+type DualWriterOptions struct {
+	Mode              DualWriterMode
```

Comment:
nitpicking here: I'm on the fence about whether we should call this `DesiredMode` instead. This is to make sure we are distinguish it from "current mode". Either way, it would be helpful to document this field and mentioned that this is the mode that the instance should transition to. I think it would be clearer for it to be called DesiredMode.

Reply:
Yes I was very confused that `grafanarest.SetDualWritingMode(...` is really a set the mode you want and it returns the mode you have